### ISSUE TITLE
Request new feature for Firth’s logistic regression

### ISSUE THREAD

**Issue by wavefancy** [JuliaStats/GLM.jl#317]
Hi Developers, 

Can I request a new feature for Firth’s logistic regression? As the GLM logistic regression has no power for (qusi-)separation data by the wald test. Firth’s logistic regression is good to compensate for this.  Ref: A Solution to Separation in Binary Response Models.

Best regards
Wallace

**Comment by nalimilan** [comment]
You're welcome to request, but this will only be implemented if somebody is willing to devote time to it. Also maybe it would better live in a separate package (not sure -- I guess it depends on whether it can benefit from reusing internal code from GLM.jl or not).

**Comment by Nosferican** [comment]
I would agree that GLM scope is limited to IRLS distribution/link regressions. More involved models are likely to be developed and hosted at a different package (e.g., [MixedModels.jl](https://github.com/dmbates/MixedModels.jl)). [Econometrics.jl](https://github.com/Nosferican/Econometrics.jl) has in its dev plans more advanced probability models... However, I would love to have a lightweight package for "safe" categorical probability models. Most of the ecosystem would likely benefit from it. The reference you mentioned is one approach. I believe the standard is to drop features/rows until it is no longer completely separable (e.g., Stata).
